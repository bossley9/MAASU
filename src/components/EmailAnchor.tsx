import React, { FC } from 'react'
import { A } from 'components/Link'

interface Props {
  children: string
  className?: string
  tabIndex?: number
}

export const EmailAnchor: FC<Props> = ({
  children,
  className = '',
  ...props
}) => {
  return (
    <A className={className} href={`mailto:${children}`} {...props}>
      {children}
    </A>
  )
}

// shorthand
export const EA = EmailAnchor
