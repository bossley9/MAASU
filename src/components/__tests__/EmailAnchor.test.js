import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { EmailAnchor } from '../EmailAnchor'

describe('render', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<EmailAnchor>test</EmailAnchor>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

it('applies the correct link subject', () => {
  const subject = 'test@test.test'
  const test = renderer.create(<EmailAnchor>{subject}</EmailAnchor>)

  expect(test.toJSON()).toMatchSnapshot()
})
