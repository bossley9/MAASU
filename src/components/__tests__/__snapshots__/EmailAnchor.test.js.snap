// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies the correct link subject 1`] = `
<a
  className="c-inh tdu hov-tdn "
  href="mailto:test@test.test"
>
  test@test.test
</a>
`;
